install/error-handling.sh

```diff
@@ -4,11 +4,36 @@ export SENTRY_DSN='https://19555c489ded4769978daae92f2346ca@self-hosted.getsentr
 export SENTRY_ORG=self-hosted
 export SENTRY_PROJECT=installer
 
-function send_event {
+function send_envelope() {
+  # Use traceback hash as the UUID since it is 32 characters long
+  local traceback_hash=$1
+  local traceback=$2
+  # escape traceback for quotes so that we are sending valid json
+  local traceback_escaped="${traceback//\"/\\\"}"
+  local envelope_file="sentry-envelope-${traceback_hash}"
+  local envelope_file_path="$basedir/$envelope_file"
+  # If the envelope file exists, we've already sent it, delete it now
+  if [[ -f envelope_file_path ]]; then
+    rm $envelope_file_path
+    return;
+  fi
+  # If we haven't sent the envelope file, make it and send to Sentry
+  # The format is documented at https://develop.sentry.dev/sdk/envelopes/
+  # Get length of file, needed for the envelope header. This is in number of characters since ▶ takes up 2 bytes and is not supported
+  local file_length=$(stat -f %z < "$basedir/$log_file")
+  # Add header for initial envelope information
+  echo '{"event_id":"'$traceback_hash'","dsn":"'$SENTRY_DSN'"}' >> $envelope_file_path
+  # Add header to specify the event type of envelope to be sent
+  echo '{"type":"event"}' >> $envelope_file_path
+  # Add traceback message to event
+  echo '{"message":"'$traceback_escaped'","level":"error"}' >> $envelope_file_path
+  # Add attachment to the event
+  echo '{"type":"attachment","length":'$file_length',"content_type":"text/plain","filename":"install_log.txt"}' >> $envelope_file_path
+  cat "$basedir/$log_file" >> $envelope_file_path
+  # Send envelope and cleanup temporary file
   local sentry_cli="docker run --rm -v $basedir:/work -e SENTRY_ORG=$SENTRY_ORG -e SENTRY_PROJECT=$SENTRY_PROJECT -e SENTRY_DSN=$SENTRY_DSN getsentry/sentry-cli"
-  command pushd .. > /dev/null
```

Comment:
I don't really see a need to do this especially since we're using $basedir in our `sentry_cli`. Maybe I'm missing something though

Reply:
Yeah this is not needed anymore.